File: Domain/Categoria.cs
using BuisnessLogic.Domain;
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain
{
    public class Categoria
    {
        public int Id { get; set; }
        public string Nombre { get; set; }
        public virtual List<PuntoTuristicoCategoria> PuntosTuristicosCategoria { get; set; }
    }
}

File: SPA/FrontEnd/src/app/Models/Categoria/categoria.ts
import { PuntoTuristicoCategoria } from '../PuntoTuristicoCategoria/punto-turistico-categoria';
export class Categoria {
    id: number;
    nombre: string;
    puntosTuristicosCategoria: PuntoTuristicoCategoria[];

    constructor(id: number, nombre: string, puntosCat: PuntoTuristicoCategoria[]) {
        this.id = id;
        this.nombre = nombre;
        this.puntosTuristicosCategoria = puntosCat;
    }
}
